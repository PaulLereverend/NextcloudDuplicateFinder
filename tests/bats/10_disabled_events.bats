#!/usr/bin/env bats
TESTSUITE="duplicatefinder"
load helper.sh

setup() {
  if [ -z "$NEXTCLOUD" ]; then
    echo "environment variable NEXTCLOUD need to bee defined"
    exit 1
  fi
  DO_NOT_SCAN=1
  load ${BATS_TEST_DIRNAME}/setup.sh
}

teardown() {
    clearTestFiles
}

@test "[$TESTSUITE] Test find-all with disabled events" {
    ./occ config:app:set --value=true duplicatefinder disable_filesystem_events   
    output=$(./occ -v duplicates:find-all -u admin)
    expectedHash="296f69b711e07c12a439e7380037e4c711fa1985d0d006b6d463c764e9580737"
    evaluateHashResult "${expectedHash}" 3 "${output}"
}

@test "[$TESTSUITE] Test upload with disabled events" {
    ./occ config:app:set --value=true duplicatefinder disable_filesystem_events   

    echo "Test" > t.txt
    curl -sS -u 'admin:admin' -T t.txt $NEXTCLOUD/remote.php/dav/files/admin/t.txt
    curl -sS -u 'admin:admin' -T t.txt $NEXTCLOUD/remote.php/dav/files/admin/t2.txt
    rm -f t.txt

    output=$(./occ -v duplicates:list -u admin)
    expectedHash="56c4396c4d371eacea44d8d183005993099d8bec360f6d43b49b4f29feda4ab2"
    evaluateHashResult "${expectedHash}" 0 "${output}"
    
    dbQuery "update oc_jobs set last_run=0,last_checked=0,reserved_at=0  where class like '%DuplicateFinder%';"
    php ./cron.php
    output=$(./occ -v duplicates:list -u admin)
    expectedHash="d243f5ebfac73b3568ea4189202b9745131f7d06756b046373f1ddc486ef92f7"
    evaluateHashResult "${expectedHash}" 4 "${output}"
}

@test "[$TESTSUITE] Test deletion with disabled events" {
    ./occ config:app:set --value=true duplicatefinder disable_filesystem_events   

    curl -sS -u 'admin:admin' -X DELETE $NEXTCLOUD/remote.php/dav/files/admin/t2.txt
    output=$(./occ -v duplicates:list -u admin)
    expectedHash="56c4396c4d371eacea44d8d183005993099d8bec360f6d43b49b4f29feda4ab2"
    evaluateHashResult "${expectedHash}" 0 "${output}"
    
    dbQuery "update oc_jobs set last_run=0,last_checked=0,reserved_at=0  where class like '%DuplicateFinder%';"
    php ./cron.php
    output=$(./occ -v duplicates:list -u admin)
    expectedHash="0a3afe9180daf73432d6dbf0c6a1bb6b215404b971a3e00f4d005f68b2272587"
    evaluateHashResult "${expectedHash}" 3 "${output}"
}
