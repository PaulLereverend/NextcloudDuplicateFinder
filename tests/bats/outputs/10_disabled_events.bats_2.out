Duplicates for user "admin" are: 
c9d04c9565fc665c80681fb1d829938026871f66e14f501e08531df66938a789(file_hash)
     /admin/files/t.txt
     /admin/files/t2.txt
1121cfccd5913f0a63fec40a6ffd44ea64f9dc135c66634ba001d10bcf4302a2(file_hash)
     /admin/files/tests/3_0.txt
     /admin/files/tests/3_1.txt
     /admin/files/tests/3_2.txt
     /admin/files/tests/3_3.txt
4355a46b19d348dc2f57c046f8ef63d4538ebb936000f3c9ee954a27460dd865(file_hash)
     /admin/files/tests/1_0.txt
     /admin/files/tests/1_1.txt
53c234e5e8472b6ac51c1ae1cab3fe06fad053beb8ebfd8977b010655bfdd3c3(file_hash)
     /admin/files/tests/2_0.txt
     /admin/files/tests/2_1.txt
     /admin/files/tests/2_2.txt
